src/Symfony/Component/Routing/Loader/AnnotationDirectoryLoader.php

```diff
@@ -76,6 +76,15 @@ function (\SplFileInfo $current) {
      */
     public function supports($resource, $type = null)
     {
+        if ($type && 'annotation' !== $type) {
```

Comment:
replaces the checks below, so they could be removed, but misses the `is_string($resource)` one